docs/misc/commands.md

```diff
@@ -0,0 +1,211 @@
+The `CommandDispatcher<S>` (`S` is often `CommandSourceStack` in this case) work like a [`DeferredRegister`](../concepts/registries.md), but it's internally created.
+Its function `register` require a `LiteralArgumentBuilder` argument.
+
+### Argument Builder
+The `ArgumentBuilder` class is abstract, so it can't be used on its own.
+It has two children:
+ - `LiteralArgumentBuilder`: `CommandDispatcher#register` only accepts it. Returned after literal arguments (one-possiblity) and executions
+ - RequiredArgumentBuilder: it has more parameters and associated get functions than `LiteralArgumentBuilder`. Returned after required arguments declarations.
+
+In fact, the entire commands are built with it.
+Last but not least, command trees are chains of builders created with some functions:
+ - `#then(ArgumentBuilder<S, ?>)`: create a new argument.
+ - `#executes(Command)`: execute the command. `Command` can be written like a `Function<CommandContext<CommandSourceStack>, Integer>` where you need to return an integer: `Command.SINGLE_SUCCESS` (or just `1`) for success and `0` otherwise.
+
+### Argument types
+Argument types is the second part of creating commands. They come in two packages:
+ - `com.mojang.brigadier.arguments` for the argument types based on [Primitive Types](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html#PageTitle) (including `String`, without `char`). Their names are always `<type>ArgumentType`.
+ - `net.minecraft.commands.arguments` for all other argument types. Their names are always `<type>Argument`.
+ - `net.neoforged.neoforge.server.command`: for mod-oriented argument like `EnumArgument`. Their names are the same as for `net.minecraft`.
+
+All these argument types are implementing an interface: `ArgumentType<T>` where `T` is the type on which they are based.
+
+When you create an argument for your commands, you will always need to choose an `ArgumentType`.
+
+### Command Contexts
+Command contexts are used every time you run a command. It's based on a generic type, the additional information, often `CommandSourceStack`.
+It contains all the additional information plus what was written as arguments.
+
+### Exceptions
+The package `com.mojang.brigadier.exceptions` provide one [exception](https://docs.oracle.com/javase/tutorial/essential/exceptions/index.html): `CommandSyntaxException`, will be thrown if the syntax is incorrect or if the command fails. It will give an error message in chat if it was a player executing the command, or as output in a [command block](https://minecraft.wiki/w/command_block) if it's a command block.
+
+It also provides many internal handler classes and the `CommandExceptionType`s.
+
+#### `CommandExceptionType`
+This is an empty interface is used to mark a class as a eponymous exception. `CommandSyntaxException` take an argument `CommandExceptionType`.
+
+The classes implementing `CommandExceptionType` aims to create exception based on a certain amount of parameters:*
```

Comment:
The exception types provide a way to take the erroring objects as an input to the failure message.